packaging/tags.py

```diff
@@ -81,35 +89,44 @@ def _cpython_interpreter(py_version):
     return "cp{major}{minor}".format(major=py_version[0], minor=py_version[1])
 
 
-def _cpython_abi(py_version):
-    soabi = sysconfig.get_config_var("SOABI")
-    if soabi:
-        options = soabi.split("-", 2)[1]
-    else:
-        found_options = [str(py_version[0]), str(py_version[1])]
-        debug = sysconfig.get_config_var("Py_DEBUG")
-        if debug or (debug is None and hasattr(sys, "gettotalrefcount")):
-            found_options.append("d")
-        if py_version < (3, 8):
-            with_pymalloc = sysconfig.get_config_var("WITH_PYMALLOC")
-            if with_pymalloc or with_pymalloc is None:
-                found_options.append("m")
-        if py_version < (3, 3):
-            unicode_size = sysconfig.get_config_var("Py_UNICODE_SIZE")
-            if unicode_size == 4 or (
-                unicode_size is None and sys.maxunicode == 0x10FFFF
-            ):
-                found_options.append("u")
-        options = "".join(found_options)
-    return "cp{options}".format(options=options)
-
-
-def _cpython_tags(py_version, interpreter, abi, platforms):
-    for tag in (Tag(interpreter, abi, platform) for platform in platforms):
-        yield tag
-    for tag in (Tag(interpreter, "abi3", platform) for platform in platforms):
+def _cpython_abis(py_version):
+    abis = []
+    version = "{}{}".format(*py_version[:2])
+    debug = pymalloc = ucs4 = ""
+    with_debug = sysconfig.get_config_var("Py_DEBUG")
+    has_refcount = hasattr(sys, "gettotalrefcount")
+    # Windows doesn't set Py_DEBUG, so checking for support of debug-compiled
+    # extension modules is the best option.
+    has_ext = "_d.pyd" in EXTENSION_SUFFIXES
+    if with_debug or (with_debug is None and (has_refcount or has_ext)):
+        debug = "d"
+    if py_version < (3, 8):
+        with_pymalloc = sysconfig.get_config_var("WITH_PYMALLOC")
+        if with_pymalloc or with_pymalloc is None:
+            pymalloc = "m"
+    if py_version < (3, 3):
```

Comment:
This conditional can be moved to be beneath `if py_version < (3, 8)` (eliminating another conditional in the main branch).